src/Context/CachedContextFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Roave\BetterReflection\Context;
+
+use Roave\BetterReflection\Context\ContextFactory;
+
+class CachedContextFactory implements ContextFactory
+{
+    private $contexts = [];
```

Comment:
can we require [PSR-16 Simple Cache](https://github.com/php-fig/simple-cache) interface for caching here, thus providing better options for caching (and also unburdening ourselves of caching mechanisms)? The only thing we'd have to take care of then is (un)serialisation.